src/Microsoft.AspNet.Razor/Generator/TagHelpers/EscapedTagHelperCodeGenerator.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Diagnostics;
+using Microsoft.AspNet.Razor.Parser.SyntaxTree;
+
+namespace Microsoft.AspNet.Razor.Generator
+{
+    /// <summary>
+    /// A <see cref="BlockCodeGenerator"/> that is responsible for generating valid 
+    /// <see cref="Compiler.LiteralChunk"/>s from escaped tag helpers.
+    /// </summary>
+    public class EscapedTagHelperCodeGenerator : SpanCodeGenerator
```

Comment:
Originally I left it out of the original Span but by doing that I'd be leaving out a character that truly existed in the document. Also, leaving out the character affected all source locations for every element following it and then got compoundingly bad the more `!`'s that were used.